File: examples/go/verify-package/verify.go
package main

import (
	"fmt"
	"os"

	statsig_go "github.com/statsig-io/statsig-go-core"
)

func main() {
	sdkKey := os.Getenv("STATSIG_SERVER_SDK_KEY")
	if sdkKey == "" {
		fmt.Println("STATSIG_SERVER_SDK_KEY is not set")
		os.Exit(1)
	}

	statsig, err := statsig_go.NewStatsig(sdkKey)
	if err != nil {
		fmt.Println("Failed to create Statsig instance: ", err)
		os.Exit(1)
	}

	statsig.Initialize()

	user, err := statsig_go.NewUserBuilderWithUserID("a_user").Build()
	if err != nil {
		fmt.Println("Failed to create Statsig user: ", err)
		os.Exit(1)
	}

	gate := statsig.CheckGate(user, "test_public")
	gcir := statsig.GetClientInitResponse(user)

	fmt.Println("-------------------------------- Get Client Initialize Response --------------------------------")
	fmt.Println(*gcir)
	fmt.Println("-------------------------------------------------------------------------------------------------")

	fmt.Println("Gate test_public: ", gate)

	// statsig.Shutdown()
}

File: examples/rust/verify-package/verify.rs
use statsig_rust::{Statsig, StatsigUser};
use std::collections::HashMap;
use std::process::Command;

fn exec(command: &str, arg: &str) -> String {
    let output = Command::new(command)
        .arg(arg)
        .output()
        .expect("Failed to execute process");
    String::from_utf8(output.stdout).expect("Failed to convert output to string")
}

#[tokio::main]
async fn main() {
    let sdk_key = std::env::var("STATSIG_SERVER_SDK_KEY").unwrap();
    if sdk_key.is_empty() {
        panic!("STATSIG_SERVER_SDK_KEY is not set");
    }

    let statsig = Statsig::new(&sdk_key, None);
    statsig.initialize().await.unwrap();

    let os = exec("uname", "-s");
    let arch = exec("uname", "-m");
    let rust_version = exec("rustc", "--version");

    println!("OS: {}", os);
    println!("Arch: {}", arch);
    println!("Rust Version: {}", rust_version);

    let mut user = StatsigUser::with_user_id("a_user");
    user.set_custom(Some(HashMap::from([
        ("os".to_string(), os),
        ("arch".to_string(), arch),
        ("rustVersion".to_string(), rust_version),
    ])));

    let gate = statsig.check_gate(&user, "test_public");
    let gcir = statsig.get_client_init_response(&user);

    println!(
        "-------------------------------- Get Client Initialize Response --------------------------------"
    );
    println!("{}", serde_json::to_string_pretty(&gcir).unwrap());
    println!(
        "-------------------------------------------------------------------------------------------------"
    );

    println!("Gate test_public: {}", gate);

    if !gate {
        panic!("\"test_public\" gate is false but should be true");
    }

    if gcir.feature_gates.is_empty()
        || gcir.layer_configs.is_empty()
        || gcir.dynamic_configs.is_empty()
    {
        panic!("GCIR is missing required fields");
    }

    println!("All checks passed, shutting down...");
    statsig.shutdown().await.unwrap();
    println!("Shutdown complete");
}
